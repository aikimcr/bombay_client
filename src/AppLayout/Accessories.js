function Accessories(props) {
    return (<div className="accessories"></div>);
}

export default Accessories;