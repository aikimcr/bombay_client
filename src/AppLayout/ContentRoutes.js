import { Routes, Route } from 'react-router-dom';

import Login from "../Login/Login";

import Top from '../Mode/Top';
import ArtistList from "../Mode/Artist/ArtistList";
import SongList from "../Mode/Song/SongList";
import PageNotFound from '../Mode/PageNotFound';

function Content(props) {
    return (
        <>
            <Routes>
                <Route path="/" element={<Top />} />
                <Route path="/artistList" element={<ArtistList />} />
                <Route path="/songList" element={<SongList />} />

                {/* Handle page not found gracefully */}
                <Route path="*" element={<PageNotFound />} />
            </Routes>
            <Login />
        </>
    );
}

export default Content;