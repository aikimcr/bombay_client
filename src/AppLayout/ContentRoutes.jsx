import { useContext } from "react";
import { Routes, Route } from "react-router-dom";

import BombayLoginContext from "../Context/BombayLoginContext";

import Login from "../Login/Login.jsx";

import Top from "../Mode/Top.jsx";
import ArtistList from "../Mode/Artist/ArtistList.jsx";
import SongList from "../Mode/Song/SongList.jsx";
import PageNotFound from "../Mode/PageNotFound.jsx";

function ContentRoutes(props) {
  const { loggedIn } = useContext(BombayLoginContext);

  function loggedOutRoutes() {
    return (
      <>
        <Route path="/" element={<Top />} />
        <Route path="/artistList" element={<Top />} />
        <Route path="/songList" element={<Top />} />
      </>
    );
  }

  function loggedInRoutes() {
    return (
      <>
        <Route path="/" element={<Top />} />
        <Route path="/artistList" element={<ArtistList />} />
        <Route path="/songList" element={<SongList />} />
      </>
    );
  }

  return (
    <>
      <Routes>
        {loggedIn ? loggedInRoutes() : loggedOutRoutes()}

        {/* Handle page not found gracefully */}
        <Route path="*" element={<PageNotFound />} />
      </Routes>
      {loggedIn ? null : <Login />}
    </>
  );
}

export default ContentRoutes;
