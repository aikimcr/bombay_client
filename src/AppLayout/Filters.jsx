function Filters(props) {
  return <div className="filters"></div>;
}

export default Filters;
