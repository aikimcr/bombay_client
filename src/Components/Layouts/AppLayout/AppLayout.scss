@use '../../../global.scss';

.app-layout {
  height: 100%;
  width: 100%;
  overflow: hidden;

  display: grid;
  grid-auto-flow: row;
  grid-template-rows: auto 1fr;
  gap: .25rem;

  background-color: var(--container-background-color);
  color: var(--container-color);

  &__header {
    --container-background-color: var(--color-peach-255);

    display: flex;
    gap: 1rem;
    justify-content: flex-start;
    align-items: center;
    flex: 0;

    @include global.standard-container;
  }

  &__content {
    display: block;

    @include global.standard-container;
  }
}