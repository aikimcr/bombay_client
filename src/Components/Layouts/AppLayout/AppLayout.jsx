import React from "react";

import "./AppLayout.scss";

export const AppLayout = ({ header, children }) => {
  return (
    <div className="app-layout">
      <header className="app-layout__header">{header}</header>
      <div className="app-layout__content">{children}</div>
    </div>
  );
};
