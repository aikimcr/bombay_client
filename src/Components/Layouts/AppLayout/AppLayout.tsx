import React, { ReactNode } from 'react';

import './AppLayout.scss';

interface AppLayoutProps {
  header: JSX.Element;
  children: ReactNode;
}

export const AppLayout: React.FC<AppLayoutProps> = ({
  header,
  children,
}): React.ReactElement => {
  return (
    <div className="app-layout">
      <header className="app-layout__header">{header}</header>
      <div className="app-layout__content">{children}</div>
    </div>
  );
};
