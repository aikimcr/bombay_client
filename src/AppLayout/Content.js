import ContentRoutes from "./ContentRoutes";

function Content(props) {
  return (
    <div className="content">
      <ContentRoutes />
    </div>
  );
}

export default Content;
