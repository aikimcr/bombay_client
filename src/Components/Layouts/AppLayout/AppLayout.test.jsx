import React from "react";
import { act, render, screen } from "@testing-library/react";
import { AppLayout } from "./AppLayout";

describe("Test AppLayout", () => {
  it("Should match snapshot", () => {
    const { asFragment } = render(
      <AppLayout header={<h1>Header</h1>}>
        <p>Content</p>
      </AppLayout>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Should wrap the header content in a header tag", () => {
    render(
      <AppLayout header={<h1 data-testid="header">Header</h1>}>
        <p>Content</p>
      </AppLayout>,
    );

    expect(screen.getByTestId("header")).toBeInTheDocument();
    expect(screen.getByTestId("header").parentElement.tagName).toEqual(
      "HEADER",
    );
    expect(screen.getByTestId("header").parentElement).toHaveClass(
      "app-layout__header",
    );
  });

  it("Should wrap the content in a div tag", () => {
    render(
      <AppLayout header={<h1>Header</h1>}>
        <p data-testid="content">Content</p>
      </AppLayout>,
    );

    expect(screen.getByTestId("content")).toBeInTheDocument();
    expect(screen.getByTestId("content").parentElement.tagName).toEqual("DIV");
    expect(screen.getByTestId("content").parentElement).toHaveClass(
      "app-layout__content",
    );
  });
});
