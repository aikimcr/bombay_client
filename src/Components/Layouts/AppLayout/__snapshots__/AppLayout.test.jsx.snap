// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AppLayout Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="app-layout"
  >
    <header
      class="app-layout__header"
    >
      <h1>
        Header
      </h1>
    </header>
    <div
      class="app-layout__content"
    >
      <p>
        Content
      </p>
    </div>
  </div>
</DocumentFragment>
`;
